src/components/common/datatable/DataTable.tsx

```diff
@@ -0,0 +1,171 @@
+"use client"
+
+import React, { type ReactNode } from "react"
+import { useDataTable } from "@/hooks/useDataTable"
+import { generateUUID } from "@/utils/common/generateUUID"
+import { ColumnSortButton } from "./ColumnSortButton"
+import { Pagination } from "./Pagination"
+import { SearchInput } from "./SearchInput"
+import type { ColumnConfig, DataTableProps } from "./types"
+
+/**
+ * Reusable data table component with configurable columns, pagination,
+ * search, and sorting capabilities.
+ *
+ * @param props - The props for the DataTable component
+ * @property columns - Configuration for table columns
+ * @property data - Array of data objects to display in the table
+ * @property [pagination] - Configuration for pagination
+ * @property [search] - Configuration for search functionality
+ * @property [sticky=true] - Whether the header row is sticky
+ * @property [emptyMessage="No data available"] - Message to display when no data is available
+ * @property [onRowClick] - Callback function when a row is clicked
+ *
+ * @example
+ * const columns: ColumnConfig<User>[] = [
+ *   { key: 'name', label: 'Name', sortable: true, searchable: true },
+ *   { key: 'email', label: 'Email', searchable: true },
+ * ];
+ *
+ * <DataTable
+ *   columns={columns}
+ *   data={users}
+ *   pagination={{ enabled: true, pageSize: 10 }}
+ *   search={{ enabled: true }}
+ * />
+ */
+export const DataTable = <T extends Record<string, unknown>>({
+  columns,
+  data,
+  pagination,
+  search,
+  sticky = true,
+  emptyMessage = "No data available",
+  onRowClick,
+}: DataTableProps<T>) => {
+  const {
+    paginatedData,
+    sortedData,
+    sortConfig,
+    currentPage,
+    handleSearchChange,
+    handleSort,
+    setCurrentPage,
+    paginationInfo,
+  } = useDataTable({
+    data,
+    columns,
+    pagination,
+    search,
+  })
+
+  const renderCellContent = (content: unknown, column: ColumnConfig<T>, row: T): ReactNode => {
+    // Use custom render function if provided
+    if (column.render) {
+      return column.render(content, row)
+    }
+
+    if (React.isValidElement(content)) {
+      return content
+    }
+    if (typeof content === "object" && content !== null) {
+      return JSON.stringify(content)
+    }
+    return String(content ?? "")
+  }
+
+  return (
+    <div className="space-y-4">
+      {/* Search Bar */}
+      {search?.enabled && <SearchInput handleSearchChange={handleSearchChange} />}
+
+      {/* Table */}
+      <div className="overflow-x-auto rounded border border-border-light">
+        <table>
```

Comment:
```suggestion
        <table className="w-full">
```
Found that this was needed or it wouldnt take up all the space if there wasnt enough rows.